### PR TITLE
BWS-35: + error handling

### PR DESCRIPTION
**Pull request by tbezrukava**

### CODE REVIEW

File: frontend/src/components/common/toastr/toastr.tsx [hunk 1]
```diff
@@ -0,0 +1,7 @@
+import { FC } from 'react';
+import ReduxToastr from 'react-redux-toastr';
+import 'react-redux-toastr/lib/css/react-redux-toastr.min.css';
+
+const Toastr: FC = () => <ReduxToastr position="top-right" />;
```
Comment (what1s1ove) [inline frontend/src/components/common/toastr/toastr.tsx]:
```suggestion
import ReduxToastr, {
  positionType,
  transitionInType,
  transitionOutType,
} from 'react-redux-toastr';

type Props = {
  timeOut?: number;
  isNewestOnTop?: boolean;
  isDuplicatesPrevented?: boolean;
  hasProgressBar?: boolean;
  isClosingOnClick?: boolean;
  position?: positionType;
  transitionIn?: transitionInType;
  transitionOut?: transitionOutType;
  className?: string;
};

const Toaster: React.FC<Props> = ({
  position = 'top-right',
}) => (
  <ReduxToastr
    timeOut={timeOut}
    newestOnTop={isNewestOnTop}
    preventDuplicates={isDuplicatesPrevented}
    progressBar={hasProgressBar}
    closeOnToastrClick={isClosingOnClick}
    position={position}
    transitionIn={transitionIn}
    transitionOut={transitionOut}
    className={className}
  />
);

```

File: frontend/src/services/notifications/notifications.tsx [hunk 2]
```diff
@@ -0,0 +1,14 @@
+import { toastr } from 'react-redux-toastr';
+import styles from './notifications.module.scss';
+
+class Notifications {
+  error = (title: string, message: string): void => {
+    const options = {
+      className: styles.wrapper,
+      icon: <div className={styles.attention} />,
+    };
+    toastr.error(title, message, options);
```
Comment (what1s1ove) [inline frontend/src/services/notifications/notifications.tsx]:
```suggestion
    toastr.error(title, message, {
      className: styles.wrapper,
      icon: <div className={styles.attention} />,
    };);
```

File: frontend/src/services/notifications/notifications.tsx [hunk 3]
```diff
@@ -0,0 +1,14 @@
+import { toastr } from 'react-redux-toastr';
+import styles from './notifications.module.scss';
+
+class Notifications {
```
Comment (what1s1ove) [inline frontend/src/services/notifications/notifications.tsx]:
```suggestion
class Notification {
```

File: frontend/src/services/notifications/notifications.tsx [hunk 4]
```diff
@@ -0,0 +1,14 @@
+import { toastr } from 'react-redux-toastr';
```
Comment (what1s1ove) [inline frontend/src/services/notifications/notifications.tsx]: rename to `frontend/src/services/notification/notification.service.tsx`

File: frontend/src/services/services.ts [hunk 5]
```diff
@@ -1,12 +1,15 @@
 import { ENV } from 'common/enums/enums';
 import { Http } from './http/http.service';
 import { AuthApi } from './auth-api/auth-api.service';
+import { Notifications } from './notifications/notifications';
 
 const http = new Http();
 
+const notifications = new Notifications();
```
Comment (what1s1ove) [inline frontend/src/services/services.ts]:
```suggestion
const notification = new Notification();
```

File: frontend/src/store/middlewares/error-handling-middleware.ts [hunk 6]
```diff
@@ -0,0 +1,16 @@
+import { Middleware } from '@reduxjs/toolkit';
+import { notifications } from '../../services/services';
+
+const errorHandlingMiddleware: Middleware =
```
Comment (what1s1ove) [inline frontend/src/store/middlewares/error-handling-middleware.ts]:
```suggestion
const handleError: Middleware =
```

File: frontend/src/store/middlewares/error-handling-middleware.ts [hunk 7]
```diff
@@ -0,0 +1,16 @@
+import { Middleware } from '@reduxjs/toolkit';
```
Comment (what1s1ove) [inline frontend/src/store/middlewares/error-handling-middleware.ts]: move to `frontend/src/store/middlewares/handle-error/handle-error.middleware.ts`

File: frontend/src/services/notifications/notifications.tsx [hunk 8]
```diff
@@ -0,0 +1,14 @@
...
+import styles from './notifications.module.scss';
+
+class Notifications {
+  error = (title: string, message: string): void => {
```
Comment (Yaroslaww-1) [inline frontend/src/services/notifications/notifications.tsx]: We strive to use `public method(param: type, ...): type` everywhere

**Review by Yaroslaww-1 — CHANGES_REQUESTED** [review]

File: frontend/src/store/root-reducer.ts [hunk 9]
```diff
@@ -1,7 +1,9 @@
 import { reducer as auth } from './auth/reducer';
+import { reducer as toastr } from 'react-redux-toastr';
```
Comment (what1s1ove) [inline frontend/src/store/root-reducer.ts:1]: import orders

File: frontend/src/store/store.ts [hunk 10]
```diff
@@ -13,7 +14,8 @@ const store = configureStore({
       thunk: {
         extraArgument,
       },
-    });
+      serializableCheck: false,
+    }).prepend(errorHandlingMiddleware);
```
Comment (what1s1ove) [inline frontend/src/store/store.ts]:
```suggestion
    }). concat(errorHandlingMiddleware);
```

**Review by what1s1ove — DISMISSED** [review]

**Review by what1s1ove — APPROVED** [review]

**Pull request merged**